ang/crmCaseType.js

```diff
@@ -276,44 +277,66 @@
+    // while the role name in the xml needs values that are names (with
+    // implicit direction).
     //
     // At any rate, the labels should follow the convention in the UI of
     // describing case roles from the perspective of the client, while the
-    // values must follow the convention in the XML of describing case roles
+    // names must follow the convention in the XML of describing case roles
     // from the perspective of the non-client.
-    function getRelationshipTypeOptions($isDefault) {
-      return _.transform(apiCalls.relTypes.values, function(result, relType) {
+    //
+    // @param onlyActive bool
+    //   If true, only include enabled relationship types.
+    // @return array[object]
+    //   object: {
+    //     xmlName: The name corresponding to what's stored in xml/caseRoles.
+    //     text: The text in dropdowns, i.e. <option value="id">text</option>
+    //       It's called text because that is what select2 is expecting.
+    //     id: The id value in dropdowns, i.e. <option value="id">text</option>
+    //       Is the concatenation of id+direction, e.g. 2_a_b.
+    //       It's called id because that is what select2 is expecting.
+    //   }
+    function getRelationshipTypeOptions(onlyActive) {
+      var relationshipTypesToUse;
+      if (onlyActive) {
+        relationshipTypesToUse = _.filter(apiCalls.relTypes.values, {is_active: "1"});
+      } else {
+        relationshipTypesToUse = apiCalls.relTypes.values;
+      }
+      return _.transform(relationshipTypesToUse, function(result, relType) {
         var isBidirectionalRelationship = relType.label_a_b === relType.label_b_a;
-        if ($isDefault) {
-          result.push({
-            label: relType.label_b_a,
-            value: relType.id + '_a_b'
-          });
 
-          if (!isBidirectionalRelationship) {
-            result.push({
-              label: relType.label_a_b,
-              value: relType.id + '_b_a'
-            });
-          }
-        }
-        // TODO The ids below really should use names not labels see
-        //  https://lab.civicrm.org/dev/core/issues/774
-        else {
+        // The order here of a's and b's here is important regarding
+        // unidirectional and bidirectional. Because the xml spec DOES support
+        // direction for activity auto-assignees, if this is changed you might
+        // end up with activity assignees in existing xml that then come
+        // through as blank in the dropdown on the timelines tab if it's a
+        // bidirectional relationship. E.g. if spouse is stored in an existing
+        // xml definition as 2_a_b, but we only push 2_b_a, then it won't match
+        // in the dropdown.
+        //
+        // This has some implications for when we add a new type on the fly
+        // later, but it works out ok as long as we also do it in the same
+        // direction there. See notes in addRoleOnTheFly().
```

Comment:
Is the unspoken assumption that all past auto-assignees have been stored as `_a_b` when relationship types are symmetrical?  Unfortunately, it looks like we may have changed that; this is how the code looks in 5.5:
https://github.com/civicrm/civicrm-core/blob/68098e7b9ee8ff39df2c602a628e2c1758c7fd89/ang/crmCaseType.js#L277-L293

That means that everyone who has defined a case type using the UI prior to 5.16 with a symmetrical relationship type has the auto-assignee set to `_b_a`.  This is probably a small set of people, but it's probably a greater number than the number who have done so since 5.16.

I think there are two ways of going about fixing this.  One option is to try to accommodate any `_b_a` values for symmetrical relationships that come in by selecting the `_a_b` equivalent.  The other is that we could do a quick fix to make it always `_b_a` in the existing code and then have you swap this PR to work with `_b_a` values, too.

Reply:
Thanks @alifrumin. Just for anyone else reading this some background on what feature this is. This isn't the roles table directly, it's on the timeline tab in the rightmost column there's a dropdown where one of the choices is "By relationship", and if you choose that you get a dropdown of relationship types. When that activity gets created, it will try to auto-assign it to the person in that role.

Soooo. I'm surprised nobody's mentioned it because you'd see it every time you create a case (unless maybe you're using it via the Add Timeline feature on manage case). And maybe it's a small number of people who use the feature in the relationship-y way and instead auto-assign to a specific contact or the case creator. And as noted, an even smaller number who've changed/created their definition since 5.16.

Anyway, yeah to fix it means maybe bumming out either group X or group Y. I don't think there's enough clues for an upgrade script to tell when it was configured or which way someone meant, except it seems very unlikely you would intend to auto-assign an activity to the client and that you'd think to do it by setting it as a reverse relationship, so that group size is even smaller. And especially if it's something like spouse, because why would you pick spouse and intend to mean the client? Maybe one way is an upgrade script that assumes you meant "not the client", and if it sees one that points to client it gives a pre-upgrade message telling you what it's going to change? I'm in upgrade-script mode right now, and that might avoid bumming anyone out.